File: conversions/typescript/Territory.ts
import { BaseElement } from './base_elements';

export interface Territory extends BaseElement {
  // Situation
  terrain?: string | null;
  size?: number | null;
  parentTerritoryId?: string | null;
  // Yield
  maintenance?: string | null;
  primaryOutput?: number | null;
  secondaryOutput?: number | null;
  primaryResourceId?: string | null;
  secondaryResourcesIds?: string[] | null;
  // World
  history?: string | null;
  occupantsIds?: string[] | null;
  occurrencesIds?: string[] | null;
}

File: languages/csharp_unity/Territory.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Territory : BaseElement
{
    // Situation
    public string Terrain;
    public int? Size;
    public string ParentTerritoryId;
    // Yield
    public string Maintenance;
    public int? PrimaryOutput;
    public int? SecondaryOutput;
    public string PrimaryResourceId;
    public List<string> SecondaryResourcesIds; = new List<string>();
    // World
    public string History;
    public List<string> OccupantsIds; = new List<string>();
    public List<string> OccurrencesIds; = new List<string>();
}
